### PR TITLE
Step-6: Add pagination

### PR DESCRIPTION
**Pull request by sta-wtag**
### Background:
Add pagination to app.

### Acceptance

- [x] Show max 9 cards initially.
- [x] Add a load more button. Upon click of the button, fetch 9 more cards and so on.
- [x] When all the items have been rendered, hide the load more button and show a show less button.
- [x] On click of show less button, show the initial 9 cards.

### CODE REVIEW

File: assets/scss/variables.scss [hunk 1]
```diff
@@ -2,3 +2,4 @@ $base-color: #f8f9ff;
 $card-color: #ffffff;
 $primaryText: #7a8dfd;
 $borderPrimary: #dde2ff;
```
Comment (hhu-wtag) [inline assets/scss/variables.scss]: Please use kebab case for consistency.

File: components/TaskContainer.vue [hunk 2]
```diff
@@ -24,14 +24,30 @@
...
+    <div class="center-item">
+      <button
+        v-if="!hasNoTask && page < totalPage"
+        class="load-button"
+        @click="loadMore"
+      >
+        {{ $t('load-more') }}
+      </button>
+      <button
+        v-if="!hasNoTask && page >= totalPage && page !== 1"
```
Comment (hhu-wtag) [inline components/TaskContainer.vue]: We can use a computed property here.

File: components/TaskContainer.vue [hunk 3]
```diff
@@ -81,6 +104,7 @@ export default {
     },
     addTask() {
       this.$store.dispatch('todos/addTask', this.taskDescription);
```
Comment (hhu-wtag) [inline components/TaskContainer.vue:104]: Can we await this if this returns promise? If it doesn't, ignore this.

File: store/todos.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
 import { uuid } from 'uuidv4';
 export const state = () => ({
+  limit: 9,
```
Comment (hhu-wtag) [inline store/todos.js]: For values that doesn't change that much but may be used in different places, we tend to create a helper method, e.g: `constants.js` and export them from there. 

This helps us reduce a lot of repetitive task once we decide to change one of these value.

File: store/todos.js [hunk 5]
```diff
@@ -1,5 +1,10 @@
...
+  totalTask: 0,
+  perPage: 9,
```
Comment (hhu-wtag) [inline store/todos.js]: Following prev review, maybe do this instead
```suggestion
  perPage: PER_PAGE,
```

Notice the constant naming.

File: store/todos.js [hunk 6]
```diff
@@ -88,4 +114,12 @@ export const mutations = {
 
     task.description = val.description;
   },
+  increaseLimit(state, val) {
+    state.page++;
+    state.perPage += state.limit;
+  },
+  resetLimit(state, val) {
+    state.perPage = 9;
```
Comment (hhu-wtag) [inline store/todos.js]: ditto
```suggestion
    state.perPage = PER_PAGE;
```

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: store/todos.js (hunk 5 above)
Comment (sta-wtag) [reply store/todos.js]: perPage is not a constant but limit is. So i can make a constants.js for Limit

File: store/todos.js (hunk 4 above)
Comment (sta-wtag) [reply store/todos.js]: Resolved

File: components/TaskContainer.vue (hunk 3 above)
Comment (sta-wtag) [reply components/TaskContainer.vue:104]: It does not return a promise just yet

File: constants.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export const Limit = 9;
```
Comment (bia-wtag) [inline constants.js]: convention: use UPPERCASE

File: store/todos.js (hunk 5 above)
Comment (bia-wtag) [reply store/todos.js]: arbitary values are problematic in the long run. use constant like `PER_PAGE` as @hhu-wtag bhai suggested as a starting point, even if `perPage` updates later.

File: store/todos.js [hunk 8]
```diff
@@ -67,11 +92,14 @@ export const mutations = {
       createdAt: new Date().toDateString(),
     };
 
-    state.taskList.push(task);
+    state.taskList = [task, ...state.taskList];
   },
   setCompleteRequest: (state, val) => {
     state.completeRequest = val;
   },
+  setTotalPage: (state, val) => {
```
Comment (bia-wtag) [inline store/todos.js]: good to prefix unused values with `_` i.e. `_val`, otherwise this'll throw error in ts.

**Review by bia-wtag — APPROVED** [review]
Tested locally. LGTM. Please have a look at the suggestions, thanks.

File: constants.js (hunk 7 above)
Comment (sta-wtag) [reply constants.js]: resolved in PR 7 .Do i need to fix it here?

File: store/todos.js (hunk 8 above)
Comment (sta-wtag) [reply store/todos.js]: noted

Comment (sta-wtag) [reply store/todos.js]: Resolved

File: constants.js (hunk 7 above)
Comment (sta-wtag) [reply constants.js]: Resolved

File: store/todos.js (hunk 5 above)
Comment (sta-wtag) [reply store/todos.js]: Resolved

**Review by sai-wtag — APPROVED** [review]
LGTM. Thanks

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**